readthedocs/donate/views.py

```diff
@@ -146,3 +150,40 @@ def view_proxy(request, promo_id, hash):
         log.warning('Duplicate view logged. {count} total clicks tried.'.format(count=count))
         cache.incr(promo.cache_key(type=VIEWS, hash=hash))
     return redirect(promo.image)
+
+
+def add_promo_data(display_type):
+    promo_queryset = SupporterPromo.objects.filter(live=True, display_type=display_type)
+    promo_obj = promo_queryset.order_by('?').first()
+    if promo_obj:
+        promo_dict = offer_promo(promo_obj=promo_obj, project=None)
+    else:
+        promo_dict = None
+    return promo_dict
+
+
+def promo_500(request, template_name='donate/promo_500.html', **kwargs):
+    """A simple 500 handler so we get media"""
+    promo_dict = add_promo_data(display_type='error')
+    r = render_to_response(template_name,
+                           context_instance=RequestContext(request),
+                           context={
+                               'promo_data': promo_dict,
+                           })
+    r.status_code = 500
+    return r
+
+
+def promo_404(request, template_name='donate/promo_404.html', **kwargs):
```

Comment:
Seems we can avoid duplication of this logic by merging `promo_500` and `promo_404`. This would be a good case for a CBV.